### PR TITLE
Bug 1462761 - update to use rootUrl

### PR DESCRIPTION
**Pull request by djmitche**
This also overhauls tc-github to follow [testing best practices](https://docs.taskcluster.net/manual/design/devel/best-practices/testing).

Note that tests won't pass until https://github.com/taskcluster/pulse-publisher/pull/29 and https://github.com/taskcluster/taskcluster-lib-api/pull/99 and https://github.com/taskcluster/taskcluster-client/pull/95 land, are relased, and the corresponding upgrades made here.

### CODE REVIEW

File: test/webhook_test.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+
+helper.secrets.mockSuite('webhook', ['taskcluster'], function(mock, skipping) {
+  helper.withEntities(mock, skipping);
+  helper.withFakeGithub(mock, skipping);
+  helper.withServer(mock, skipping);
 
   let github = null;
 
-  setup(async () => {
+  setup(async function() {
```
Comment (owlishDeveloper) [inline test/webhook_test.js:11]: why ditch arrow syntax?

**Review by owlishDeveloper — APPROVED** [review]

File: test/webhook_test.js (hunk 1 above)
Comment (djmitche) [reply test/webhook_test.js:11]: The mocha functions all pass a context that is useful for things like `this.pending = true`, and to get a context you need to use `function` and not an arrow.  Most of the time it doesn't matter, but it's easier to just be consistent about it.

Comment (imbstack) [reply test/webhook_test.js:11]: @owlishDeveloper it is just a funny mocha-specific thing https://mochajs.org/#arrow-functions

Comment (owlishDeveloper) [reply test/webhook_test.js:11]: wow, I didn't know this! ok  👍

**Review by imbstack — APPROVED** [review]
This is great. Note that you will need to rebase now since I landed @owlishDeveloper changes. 😅

**Comment by owlishDeveloper** [comment]
😊 😇

**Comment by djmitche** [comment]
I've upgraded this to use pulse-publisher 10.0.0 as well.  I'd ask for another look, but there's really not much to see, so unless there's objection I'll probably land as-is, with the existing :heavy_check_mark:

**Comment by djmitche** [comment]
I can't reproduce those CI failures, even in a local `node:8` docker container...

**Comment by djmitche** [comment]
Oh, it's because they're running in parallel and use the same Azure tables, I think.

**Pull request merged**